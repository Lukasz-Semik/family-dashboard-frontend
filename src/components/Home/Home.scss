.title-wrapper {
  margin-left: 1rem;
  margin-top: 0.3rem;
}
