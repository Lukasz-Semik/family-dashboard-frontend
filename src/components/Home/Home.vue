<template>
  <div>
    <div :class="[$style['title-wrapper']]">
      <Title translationPath="general.appTitle"></Title>
    </div>

    <router-view></router-view>
  </div>
</template>

<script>
import Title from '@/components/atoms/Title/Title.vue';

export default {
  components: {
    Title,
  },
};
</script>

<style lang="scss" module src="./Home.scss" />
