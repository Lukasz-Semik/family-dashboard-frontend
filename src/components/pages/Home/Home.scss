.title-wrapper {
  margin: 0.3rem 0 2rem 1rem;
}

.links-wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 60rem;
  max-width: 100%;
  margin: 0 auto 1rem;
}

.toggle-view-leave-active,
.toggle-view-enter-active {
  transition: opacity 1s ease;
}

.toggle-view-leave,
.toggle-view-enter {
  opacity: 0;
}
