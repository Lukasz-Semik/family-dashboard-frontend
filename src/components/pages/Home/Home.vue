<template>
  <div>
    <div :class="[$style['title-wrapper']]">
      <Title translation-path="general.appTitle" />
    </div>

    <div :class="[$style['links-wrapper']]">
      <Link
        :to="signInRoute"
        translation-path="general.signIn"
      />
      <Link
        :to="signUpRoute"
        translation-path="general.signUp"
      />
    </div>

    <Transition
      :leave-active-class="$style['toggle-view-leave-active']"
      :leave-class="$style['toggle-view-leave']"
      :enter-active-class="$style['toggle-view-enter-active']"
      :enter-class="$style['toggle-view-enter']"
      mode="out-in"
    >
      <RouterView />
    </Transition>
  </div>
</template>

<script>
import Title from '@/components/atoms/Title/Title.vue';
import Link from '@/components/atoms/Link/Link.vue';
import { SIGN_IN_ROUTE, SIGN_UP_ROUTE } from '@/constants/routesNames';

export default {
  components: {
    Title,
    Link,
  },
  data() {
    return {
      signInRoute: { name: SIGN_IN_ROUTE },
      signUpRoute: { name: SIGN_UP_ROUTE },
    };
  },
};
</script>

<style lang="scss" module src="./Home.scss" />
