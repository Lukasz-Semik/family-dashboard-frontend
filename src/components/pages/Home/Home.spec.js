import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';

import Home from './Home';

Vue.use(VueRouter);

describe('<Home/>', () => {
  it('should render properly', () => {
    const wrapper = shallowMount(Home);

    expect(wrapper).toMatchSnapshot();
  });
});
