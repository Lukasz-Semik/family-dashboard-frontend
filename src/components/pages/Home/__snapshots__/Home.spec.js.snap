// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should render properly 1`] = `
<div>
  <div class="title-wrapper">
    <title-stub tag="h1" translatedtext="" translationpath="general.appTitle"></title-stub>
  </div>
  <div class="links-wrapper">
    <link-stub to="[object Object]" translatedtext="" translationpath="general.signIn"></link-stub>
    <link-stub to="[object Object]" translatedtext="" translationpath="general.signUp"></link-stub>
  </div>
  <routerview-stub name="default"></routerview-stub>
</div>
`;
